main/ejml-core/src/org/ejml/UtilEjml.java

```diff
@@ -574,35 +577,44 @@ public static String fancyString(double value, DecimalFormat format, boolean has
     /**
      * Returns true if any of the matrix arguments has @Nullable
      */
-    public static boolean hasNullableArgument(Method func) {
+    public static boolean hasNullableArgument( Method func ) {
         Annotation[][] annotations = func.getParameterAnnotations();
-        if( annotations.length == 0)
+        if (annotations.length == 0)
             return false;
 
         Class<?>[] types = func.getParameterTypes();
         for (int i = 0; i < types.length; i++) {
             Annotation[] argumentAnnotations = annotations[i];
-            if( argumentAnnotations.length == 0)
+            if (argumentAnnotations.length == 0)
                 continue;
-            if( !Matrix.class.isAssignableFrom(types[i]))
+            if (!Matrix.class.isAssignableFrom(types[i]))
                 continue;
-            Annotation last = argumentAnnotations[argumentAnnotations.length-1];
-            if( last.toString().contains("Nullable") )
+            Annotation last = argumentAnnotations[argumentAnnotations.length - 1];
+            if (last.toString().contains("Nullable"))
                 return true;
         }
         return false;
     }
 
     public static GrowArray<FGrowArray> checkDeclare_F32( @Nullable GrowArray<FGrowArray> workspace ) {
-        if( workspace == null)
+        if (workspace == null)
             return new GrowArray<>(FGrowArray::new);
         return workspace;
     }
 
     public static GrowArray<DGrowArray> checkDeclare_F64( @Nullable GrowArray<DGrowArray> workspace ) {
-        if( workspace == null)
+        if (workspace == null)
             return new GrowArray<>(DGrowArray::new);
         return workspace;
     }
 
+    /**
+     * Checks to see if a matrix of this size will exceed the maximum possible value an integer can store, which is
+     * the max possible array size in Java.
+     */
+    public static boolean exceedsMaxMatrixSize( int numRows, int numCols ) {
+        if (numRows == 0 || numCols == 0)
+            return false;
+        return numCols > Integer.MAX_VALUE/numRows;
```

Comment:
As mentioned in other comments, I think this check is valid for the current random matrix creation.
However in general a sparse random matrix can have at maximum `numRows == Integer.MAX_VALUE` and `numCols == Integer.MAX_VALUE`. The important thing is, that the nz-values do not exceed this limit.

Reply:
Isn't that what this is checking? Mathematically this is identical to `numCols*numRows > Integer.MAX_VALUE` but takes in account the possibility of `numCols*numRows` overflowing.

Reply:
Never mind I had a brain fart. Fixing all of this.